### PR TITLE
Take out the cycle between federation and others

### PR DESCRIPTION
**Pull request by aiuto**
We can not have any load() methods in repositories.bzl that reach outside of the federation because then people have to download and load even if they do not use the rule,.

Fix examples/stardoc
fix examples/rules_pkg
fix test/integegration/hello_bazel
fix .bazelci/rules_javaa

I did not get to test go.

### CODE REVIEW

**Comment by aiuto** [comment]
Goes with https://github.com/bazelbuild/rules_java/pull/14

File: .bazelci/rules_java.yml [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
     setup:
     - python3.6 create_project_workspace.py --project=rules_java
   ubuntu1804:
     build_targets:
-    - '@rules_java//...'
+    - '@rules_java//java/...'
     setup:
     - python3.6 create_project_workspace.py --project=rules_java
-  ubuntu1804_nojava:
+  ubuntu1804_nojava/
```
Comment (fweikert) [inline .bazelci/rules_java.yml:17]: Syntax error: revert change (there should be a colon, not a slash)

File: repositories.bzl [hunk 2]
```diff
@@ -137,7 +118,7 @@ def protobuf_javalite():
 def rules_cc_deps():
     bazel_skylib()
 
-def rules_cc():
+def XXrules_cc():
```
Comment (fweikert) [inline repositories.bzl:121]: ?

File: repositories.bzl [hunk 3]
```diff
@@ -106,7 +87,7 @@ def bazel_stardoc():
 
 # TODO(fweikert): delete this function if it's not needed by the protobuf project itself.
 def protobuf_deps():
-    zlib()
+    # zlib()
```
Comment (fweikert) [inline repositories.bzl:90]: Why did you remove zblib?

File: repositories.bzl [hunk 4]
```diff
@@ -166,9 +159,69 @@ def rules_go():
         sha256 = "f04d2373bcaf8aa09bccb08a98a57e721306c8f6043a2a0ee610fd6853dcde3d",
     )
 
+
+def org_golang_x_tools():
```
Comment (fweikert) [inline repositories.bzl:163]: This and the following functions are already part of third_party_repositories.bzl

File: repositories.bzl [hunk 5]
```diff
@@ -166,9 +159,69 @@ def rules_go():
...
+        name = "com_github_bazelbuild_buildtools",
+        strip_prefix = "buildtools-0.20.0",
+        url = "https://github.com/bazelbuild/buildtools/archive/0.20.0.zip",
+        sha256 = "7b46f95f1df24a62dbe1953cb7820f4170525f72b93a7f9fe414e027a3151128",
+    )
+
 def rules_java_deps():
-    remote_jdk11_repos()
-    java_tools_javac11_repos()
+    # TODO(aiuto): Java dependencies are currently brought in through
```
Comment (fweikert) [inline repositories.bzl:222]: I don't understand - why should we revert previous work?

File: repositories.bzl [hunk 6]
```diff
@@ -194,12 +247,48 @@ def rules_nodejs():
...
+
+# The @federation markers are an experiment in how to pick up dependency stanzas
+# from the rule sets. Each rule set should have a deps.bzl file. Inside those,
+# there will be the @federation markers. Those must be designed so they can
+# be extracted and dropped directly into this file, replacing the same stanza.
+
+# @federation: BEGIN @rules_pkg
+# TODO: ptr to where this was extracted from
+# https://github.com/bazelbuild/rules_pkg/pkg/deps.bzl
+def abseil_py():
```
Comment (fweikert) [inline repositories.bzl:259]: This function and the following ones are already in third_party_repositories.bzl

File: repositories.bzl [hunk 7]
```diff
@@ -166,9 +159,69 @@ def rules_go():
...
+    org_golang_x_tools()
+    org_golang_x_sys()
+    http_archive(
+        name = "io_bazel_rules_go",
+        sha256 = "7be7dc01f1e0afdba6c8eb2b43d2fa01c743be1b9273ab1eaf6c233df078d705",
+        urls = ["https://github.com/bazelbuild/rules_go/releases/download/0.16.5/rules_go-0.16.5.tar.gz"],
+    )
+
+
+def bazel_gazelle_deps():
```
Comment (fweikert) [inline repositories.bzl:200]: I think you need to pull my previous changes and merge them, since I ordered all functions by name, and this commit reverts that order for no real reason.

**Review by fweikert — CHANGES_REQUESTED** [review]
I don't understand the intent of this PR - the load statements in repositories.bzl bring in other bzl files that are in the federation repo, too. The functions in these files do refer to other repositories, but those are only loaded if the functions are actually invoked.

**Comment by fweikert** [comment]
Friendly ping

**Comment by aiuto** [comment]
I was trying to do too many different things in this CL. I'll send individual ones we can debate about.
